alica_engine/include/engine/modelmanagement/factories/BlackboardFactory.h

```diff
@@ -0,0 +1,19 @@
+#pragma once
+
+#include "Factory.h"
+
+#include <yaml-cpp/yaml.h>
+
+namespace alica
+{
+class Blackboard;
+class BlackboardFactory : public Factory
+{
+public:
+    static std::unique_ptr<Blackboard> create(const YAML::Node& node);
```

Comment:
Can we just return a shared_ptr here?  Returning a unique ptr then moving it into a shared ptr is kind of weird